### ISSUE TITLE
[Feature Request] Add more options to the optimizer

### ISSUE THREAD

**Issue by marcociccone** [fvisin/main_loop_tf#8]
I will need all of the features of this optimizer: 

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/optimizers.py

We can both integrate it inside the main loop, or just take the things we need

**Comment by fvisin** [comment]
You linked the whole file, not a specific optimizer.
Anyway, it's already possible to specify as many options as you want through `optimizer_params` (which can be called more than once, e.g., `python run.py --optimizer_params="epsilon=0.5" --optimizer_params="learning_rate=1e-05"`)

**Comment by marcociccone** [comment]
ok thanks, I refer to other options such as gradient clipping, gradient noise etc.
The function that I meant is `optimize_loss` inside the file, the one we briefly discussed.

**Comment by fvisin** [comment]
Was it fixed in 411e14e40de0de45e24ab1383216ce95da3133e5 or is something missing?

**Comment by marcociccone** [comment]
yes I'll close it.
